### PR TITLE
Add seed recovery component logic

### LINKED ISSUE #183: "Reset your password" put different phrases - logic
image

### PR DESCRIPTION
**Pull request by banciur**
This PR is mostly about logic of seed recovery component. For now its available under `/register/light/seed` route but we need to put it in correct one and add proper view

screencapture-localhost-9090-register-light-seed-1519297094309
screencapture-localhost-9090-register-light-seed-1519297114858

closes #183

### CODE REVIEW

File: app/components/walletSelector/walletLight/WalletLightSeedRecovery.tsx [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+
+    return (
+      <>
+        <WalletResetHeader
+          text={"Use the Recovery Phrase to restore your password."}
+          currentStep={this.state.page + 1}
+          steps={SEED_LENGTH / WORDS_PER_VIEW}
+        />
+        <Row className="justify-content-center my-3">
+          <div>{this.state.words.filter(word => word !== null).join(" , ")}</div>
```
Comment (Moejoe90) [inline app/components/walletSelector/walletLight/WalletLightSeedRecovery.tsx]: Text is not in the center you can use `text-center`

File: app/components/walletSelector/walletLight/WalletLightSeedRecovery.tsx [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+    const canSubmit = this.state.words.reduce((acc: boolean, value: string): boolean => {
+      return acc && value !== null;
+    }, true);
+
+    return (
+      <>
+        <WalletResetHeader
+          text={"Use the Recovery Phrase to restore your password."}
+          currentStep={this.state.page + 1}
+          steps={SEED_LENGTH / WORDS_PER_VIEW}
```
Comment (Moejoe90) [inline app/components/walletSelector/walletLight/WalletLightSeedRecovery.tsx]: steps={(SEED_LENGTH / WORDS_PER_VIEW)+1}
For the last succssess page

File: webpack/webpack.config.dev.js [hunk 3]
```diff
@@ -84,6 +84,23 @@ module.exports = merge(configCommon, {
               { loader: "sass-loader" },
             ],
           },
+          {
```
[the remark this answers is not in the document]
Comment (banciur) [reply webpack/webpack.config.dev.js]: Yes,  in `WalletLightSeedRecovery.tsx` I had to import some css files to make autocomplete plugin work. 
`import "react-select/dist/react-select.css";`
For this we need to define new wepack loading rule.

File: app/components/walletSelector/walletLight/WalletLightSeedRecovery.tsx (hunk 1 above)
Comment (banciur) [reply app/components/walletSelector/walletLight/WalletLightSeedRecovery.tsx]: fixed

File: webpack/webpack.config.dev.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (Moejoe90) [reply webpack/webpack.config.dev.js]: I see

File: app/components/walletSelector/walletLight/WalletLightSeedRecovery.tsx (hunk 2 above)
Comment (banciur) [reply app/components/walletSelector/walletLight/WalletLightSeedRecovery.tsx]: As we talked I added properties `startingStep` and `extraSteps`

File: app/components/walletSelector/walletLight/WalletLightSeedRecovery.tsx [hunk 4]
```diff
@@ -0,0 +1,148 @@
+import * as Mnemonic from "bitcore-mnemonic";
+import { range } from "lodash";
+import * as React from "react";
+/* tslint:disable: no-submodule-imports */
+import "react-select/dist/react-select.css";
```
Comment (krzkaczor) [inline app/components/walletSelector/walletLight/WalletLightSeedRecovery.tsx]: could you group css imports together at least?

File: app/components/walletSelector/walletLight/WalletLightSeedRecovery.spec.tsx [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+  it("buttons should be disabled for initial state", () => {
+    const component = shallow(<WalletLightSeedRecoveryComponent {...defaultProps()} />);
+    expect(component.find(tid("btn-previous")).prop("disabled")).to.be.true;
+    expect(component.find(tid("btn-next")).prop("disabled")).to.be.true;
+    expect(component.find(tid("btn-send")).prop("disabled")).to.be.true;
+  });
+
+  it("previous button should be active on pages higher than 0 ", () => {
+    const component = shallow(<WalletLightSeedRecoveryComponent {...defaultProps()} />);
+    const state = initialState();
```
Comment (krzkaczor) [inline app/components/walletSelector/walletLight/WalletLightSeedRecovery.spec.tsx]: or you could write something like `const state = { ...initialState(), page: 1 };`

File: app/components/walletSelector/walletLight/WalletLightSeedRecovery.spec.tsx [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+describe("<WalletLightSeedRecoveryComponent />", () => {
+  it("buttons should be disabled for initial state", () => {
+    const component = shallow(<WalletLightSeedRecoveryComponent {...defaultProps()} />);
+    expect(component.find(tid("btn-previous")).prop("disabled")).to.be.true;
+    expect(component.find(tid("btn-next")).prop("disabled")).to.be.true;
+    expect(component.find(tid("btn-send")).prop("disabled")).to.be.true;
+  });
+
+  it("previous button should be active on pages higher than 0 ", () => {
+    const component = shallow(<WalletLightSeedRecoveryComponent {...defaultProps()} />);
```
Comment (krzkaczor) [inline app/components/walletSelector/walletLight/WalletLightSeedRecovery.spec.tsx]: could you improve redablitiy of these tests? First gather data, then act (render, set state) then assert. All of these should be separated with new lines.

File: app/components/walletSelector/walletLight/WalletLightSeedRecovery.spec.tsx [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+    component.find(tid("btn-next")).simulate("click");
+    expect(component.state().page).to.be.eq(1);
+  });
+
+  it("send button should call correct method", () => {
+    const props = defaultProps();
+    const component = shallow(<WalletLightSeedRecoveryComponent {...props} />);
+    component.find(tid("btn-send")).simulate("click");
+    expect(props.sendWords).to.be.called;
```
Comment (krzkaczor) [inline app/components/walletSelector/walletLight/WalletLightSeedRecovery.spec.tsx]: assert that proper data was sent.

File: app/components/walletSelector/walletLight/WalletLightSeedRecovery.spec.tsx [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+
+  it("previous button should call method to increase page number", () => {
+    const component = shallow(<WalletLightSeedRecoveryComponent {...defaultProps()} />);
+    component.find(tid("btn-next")).simulate("click");
+    expect(component.state().page).to.be.eq(1);
+  });
+
+  it("send button should call correct method", () => {
+    const props = defaultProps();
+    const component = shallow(<WalletLightSeedRecoveryComponent {...props} />);
```
Comment (krzkaczor) [inline app/components/walletSelector/walletLight/WalletLightSeedRecovery.spec.tsx]: btw. i think it shouldn't be possible to send words if they are not set oO maybe this test is not written correctly.

File: app/components/walletSelector/walletLight/WalletLightSeedRecovery.tsx (hunk 4 above)
Comment (banciur) [reply app/components/walletSelector/walletLight/WalletLightSeedRecovery.tsx]: fixed

**Pull request merged**